MASFoundation/Classes/_private_/services/model/MASModelService.m

```diff
@@ -1991,6 +1993,30 @@ - (void)loginAsRefreshTokenWithCompletion:(MASCompletionErrorBlock)completion
                                           //
                                           if (error)
                                           {
+                                              //
+                                              // DE509848 - During server maintenance timeframe,
+                                              // when token refresh calls returns 500, MAS library logout the user.
+                                              //
+                                              if ([MAS isDonotLogoutTokenRenewalOnServerErrors] && httpResponse.statusCode == 500)  {
+                                                  
+                                                  NSError *idTokenValidationError = nil;
+                                                  BOOL isIdTokenValid = [MASAccessService validateIdToken:[[MASAccessService sharedService] getAccessValueStringWithStorageKey:MASKeychainStorageKeyIdToken]
+                                                                                            magIdentifier:[[MASAccessService sharedService] getAccessValueStringWithStorageKey:MASKeychainStorageKeyMAGIdentifier]
+                                                                                                    error:&idTokenValidationError];
+                                                  
+                                                  NSString *refreshToken =
```

Comment:
I think we've not been counting on the refresh-token. Please check: https://github.com/CAAPIM/Android-MAS-SDK/pull/397/files#diff-4b5920088a9b497400ba73a15a524ed770fa6749f764fd9a20d5a786060e5bc4R261-R267